sphinx_gallery/gen_rst.py

```diff
@@ -543,6 +543,15 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
 
     src_file = os.path.normpath(os.path.join(src_dir, fname))
     example_file = os.path.join(target_dir, fname)
+    binder_fpath = '_downloads/{}'.format(fname.replace('.py', '.ipynb'))
+    binder_url = gallery_conf['binder_url']
+
+    if binder_url is not None:
+        binder_url = '/'.join([binder_url, 'v2', 'gh',
+                               gallery_conf['binder_org'],
+                               gallery_conf['binder_repo'],
+                               gallery_conf['binder_branch']])
+        binder_url += '?filepath={}'.format(binder_fpath)
```

Comment:
Can you move this to a separate function. That way we can also test for it. Also you define this here and use `binder_url` a 100 lines later. Make it so that the function returns empty string if Binder in not configured or the rst to put the badge in the example and call it when we are writing the file.